pkg/ec2pricing/ec2pricing.go

```diff
@@ -154,25 +184,27 @@ func (p *EC2Pricing) GetOndemandInstanceTypeCost(instanceType string) (float64,
 		},
 	}
 
-	// Check cache first and return it if available
-	if price, ok := p.cache[instanceType]; ok {
-		return price, nil
-	}
-
 	pricePerUnitInUSD := float64(-1)
-	err := p.PricingClient.GetProductsPages(&productInput, func(pricingOutput *pricing.GetProductsOutput, nextPage bool) bool {
-		var err error
+	var processingErr error
+	// FIXME Only works for us-east-1 and ap-south-1
+	// https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/using-pelong.html#pe-endpoint
+	errAPI := p.PricingClient.GetProductsPages(&productInput, func(pricingOutput *pricing.GetProductsOutput, nextPage bool) bool {
+		var errParse error
 		for _, priceDoc := range pricingOutput.PriceList {
-			_, pricePerUnitInUSD, err = parseOndemandUnitPrice(priceDoc)
+			_, pricePerUnitInUSD, errParse = parseOndemandUnitPrice(priceDoc)
 		}
-		if err != nil {
+		if errParse != nil {
```

Comment:
I realize you didn't change this part... but it seems like this `if` should go in the for loop to catch all parse errors.

Reply:
Good catch!